### PR TITLE
Backport Beta 4 fixes to WordPress Core.

### PR DESCRIPTION
**Pull request by youknowriad**
This includes:

Beta 4

Various

 - JS error when pasting multiple times into a RichText component. https://github.com/WordPress/gutenberg/pull/32527
 - No direct import of @wordpress/editor in @wordpress/block-library. https://github.com/WordPress/gutenberg/pull/32866
 - Fix newly created template not appearing in the template selector. https://github.com/WordPress/gutenberg/pull/32744
 - Fix collapsing appender when the paragraph block is disabled. https://github.com/WordPress/gutenberg/pull/32894
 - Correctly set the image size slug. https://github.com/WordPress/gutenberg/pull/32364
 - Avoid flash of background when scrolling in Safari. https://github.com/WordPress/gutenberg/pull/32747
 - https://github.com/WordPress/gutenberg/pull/32814

Widget editor

 - Don’t display admin notices in the widgets screen. https://github.com/WordPress/gutenberg/pull/32877
 - Fix scroll jitter in the block-based customizer widgets. https://github.com/WordPress/gutenberg/pull/32479
 - Fix Legacy widget block editor style bleed. https://github.com/WordPress/gutenberg/pull/32871
 - Stretch the widgets editor screen to full height by default. https://github.com/WordPress/gutenberg/pull/32905
 - Disable FSE blocks in the widgets editor. https://github.com/WordPress/gutenberg/pull/32761

### CODE REVIEW

**Comment by youknowriad** [comment]
Looks like Github now expands the title of the PRs, no need to write them manually anymore :)

**Comment by oandregal** [comment]
@youknowriad are we still on time to get https://github.com/WordPress/gutenberg/pull/31840 in? Trying to make tests pass.

File: webpack.config.js [hunk 1]
```diff
@@ -138,14 +141,17 @@ module.exports = {
...
+			const request = rootModule?.rawRequest || rawRequest;
 
-			/*
-			 * If the file being built is a Core Block's frontend file,
-			 * we build it in the block's directory.
-			 */
-			if ( rawRequest && rawRequest.includes( '/frontend.js' ) ) {
-				return `./build/block-library/blocks/[name]/frontend.js`;
+			if ( request.includes( '/view.js' ) ) {
+				return `./build/block-library/[name]/view.min.js`;
```
Comment (gziolo) [inline webpack.config.js]: I think we don't have the changes from the `trunk` that uses `.min.js` extension:

```suggestion
				return `./build/block-library/[name]/view.js`;
```

Related PR: https://github.com/WordPress/gutenberg/pull/31732.

**Comment by youknowriad** [comment]
@nosolosw yes, we can wait for it.

File: lib/blocks.php [hunk 2]
```diff
@@ -164,13 +164,39 @@ function gutenberg_reregister_core_block_types() {
...
  * @return void
  */
-function gutenberg_register_core_block_styles( $block_name ) {
+function gutenberg_register_core_block_assets( $block_name ) {
+	$block_name = str_replace( 'core/', '', $block_name );
+
+	// When in production, use the plugin's version as the default asset version;
+	// else (for development or test) default to use the current time.
+	$default_version = defined( 'GUTENBERG_VERSION' ) && ! ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ? GUTENBERG_VERSION : time();
+	$script_suffix   = '.min.js';
```
Comment (gziolo) [inline lib/blocks.php]:
```suggestion
	$script_suffix   = '.js';
```

File: packages/e2e-tests/specs/experiments/blocks/navigation.test.js [hunk 3]
```diff
@@ -559,7 +559,7 @@ describe( 'Navigation', () => {
 			() =>
 				null !==
 				document.querySelector(
-					'script[src*="navigation/frontend.js"]'
+					'script[src*="navigation/view.min.js"]'
```
Comment (gziolo) [inline packages/e2e-tests/specs/experiments/blocks/navigation.test.js]:
```suggestion
					'script[src*="navigation/view.js"]'
```

File: packages/e2e-tests/specs/experiments/blocks/navigation.test.js [hunk 4]
```diff
@@ -582,7 +582,7 @@ describe( 'Navigation', () => {
 			() =>
 				Array.from(
 					document.querySelectorAll(
-						'script[src*="navigation/frontend.js"]'
+						'script[src*="navigation/view.min.js"]'
```
Comment (gziolo) [inline packages/e2e-tests/specs/experiments/blocks/navigation.test.js]:
```suggestion
						'script[src*="navigation/view.js"]'
```

File: packages/e2e-tests/specs/experiments/blocks/navigation.test.js [hunk 5]
```diff
@@ -614,7 +614,7 @@ describe( 'Navigation', () => {
 			() =>
 				null !==
 				document.querySelector(
-					'script[src*="navigation/frontend.js"]'
+					'script[src*="navigation/view.min.js"]'
```
Comment (gziolo) [inline packages/e2e-tests/specs/experiments/blocks/navigation.test.js]:
```suggestion
					'script[src*="navigation/view.js"]'
```

File: packages/e2e-tests/specs/experiments/blocks/navigation.test.js [hunk 6]
```diff
@@ -632,7 +632,7 @@ describe( 'Navigation', () => {
...
```
Comment (gziolo) [inline packages/e2e-tests/specs/experiments/blocks/navigation.test.js]:
```suggestion
					'script[src*="navigation/view.js"]'
```

**Comment by gziolo** [comment]
I have just tested the File block and PDF previews on the frontend and it works as expected 🎉

**Pull request merged**